File: app/src/main/java/com/pixels/blockies/app/game/figures/FigureZ.java
package com.pixels.blockies.app.game.figures;

import com.pixels.blockies.app.game.Grid;

public class FigureZ extends AbstractFigure {

    private int[][][] figures = new int[][][]{
            {
                    {5, Grid.EMPTY},
                    {5, 5},
                    {Grid.EMPTY, 5}
            },
            {
                    {Grid.EMPTY, 5, 5},
                    {5, 5, Grid.EMPTY}
            }
    };

    @Override
    protected int getFigureCount() {
        return figures.length;
    }

    @Override
    public int[][] get() {
        return figures[getCurrentRotation()];
    }

    @Override
    public int[][] getNext() {
        return figures[getNextRotation()];
    }
}

File: app/src/main/java/com/pixels/blockies/game/game/figures/FigureZ.kt
package com.pixels.blockies.game.game.figures

import com.pixels.blockies.game.game.Grid

class FigureZ : AbstractFigure() {
    private val figures = arrayOf(
        arrayOf<IntArray?>(
            intArrayOf(5, Grid.Companion.EMPTY),
            intArrayOf(5, 5),
            intArrayOf(Grid.Companion.EMPTY, 5)
        ),
        arrayOf<IntArray?>(
            intArrayOf(Grid.Companion.EMPTY, 5, 5),
            intArrayOf(5, 5, Grid.Companion.EMPTY)
        )
    )

    override fun getFigureCount(): Int {
        return figures.size
    }

    override fun get(): Array<IntArray?> {
        return figures[currentRotation]
    }

    override fun getNext(): Array<IntArray?> {
        return figures[nextRotation]
    }
}